ERCS/erc-7994.md

```diff
@@ -0,0 +1,198 @@
+author: Anushka Yadav (@64anushka), Akash Kothawade (@akash3927), Atishek Singh (@atisheksingh)
+discussions-to: https://ethereum-magicians.org/t/erc-7994-purpose-bound-erc20-with-multi-condition-unlocking-extension-of-eip-7291/24945
+status: Draft
+type: Standards Track
+category: ERC
+created: 2025-07-29
+---
+
+## Abstract
+
+This ERC extends the concept introduced in EIP-7291 by enabling ERC20-compatible tokens to carry multi-condition unlocking constraints, combining temporal, identity, and usage restrictions into a programmable structure. It aims to support controlled disbursement of tokens where funds are only accessible under predefined, auditable, and verifiable conditions.
+
+## Motivation
+
+EIP-7291 introduced purpose-bound money by restricting how and where tokens can be spent. However, many real-world applications require multiple conditions to be satisfied simultaneously before tokens can be used. Examples include:
+
+Scholarships requiring the recipient to be KYC-verified, under 25 years of age, and registered at a university.
+
+NGO aid to be used only for food and medicine, after a specific unlock date.
+
+Payroll tokens that unlock monthly and only for whitelisted vendors (e.g., banks, healthcare providers).
+
+This proposal generalizes and formalizes such use cases by layering unlocking conditions on top of the ERC20 standard.
+
+## Specification
+
+The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “NOT RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+### Interface
+
+```solidity
+
+pragma solidity 0.8.23;
+
+interface IPurposeBoundERC20 {
+    struct UnlockCondition {
+        bytes32 conditionType; // e.g., "TIME", "KYC", "WHITELIST"
+        bytes conditionData;   // e.g., timestamp, Merkle root, etc.
+    }
+
+    struct PurposeBinding {
+        address recipient;
+        uint256 amount;
+        UnlockCondition[] conditions;
+        uint256 expiry; // optional expiration of binding
+    }
+
+    function bindPurpose(
+        address recipient,
+        uint256 amount,
+        UnlockCondition[] calldata conditions,
+        uint256 expiry
+    ) external returns (bytes32 bindingId);
+
+    function claim(bytes32 bindingId) external;
+
+    function isUnlocked(bytes32 bindingId) external view returns (bool);
+}
+```
+
+## Rationale
+
+Flexibility: Conditions are modular and extensible.
+
+Composability: Can be integrated into DAOs, payroll, education, and compliance tokens.
+
+Security: Off-chain verification (e.g., KYC) backed by on-chain proofs (e.g., Merkle roots).
+
+## Reference Implementation
+
+```solidity
+// SPDX-License-Identifier: MIT
```

Comment:
Code within the document itself must be CC0-1.0 licensed. If you'd like to keep this MIT, please move it into the assets directory and link to it there.